app/Http/Controllers/AuthorizedUsers/DashboardController.php

```diff
@@ -6,20 +6,49 @@
 use Illuminate\Http\Request;
 use App\Models\StudentRegister;
 use App\Models\Classe;
+use App\Models\Teacher;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\Facades\Auth;
 
 class DashboardController extends Controller
 {
     /**
      * Display a listing of the resource.
      */
-    public function index()
+    public function index(Request $request)
     {
-        $students = StudentRegister::all();
-        $classes = Classe::all();
+        $userId = Auth::id();
 
-        return view(('teacher.presents'), compact('students', 'classes'));
+        $teacher = Teacher::where('user_id', $userId)->first();
+
+        if ($teacher) {
```

Comment:
gestire il caso in cui mancano teachers o students, in quel caso dare in input due array vuoti così:

$students = [];
$classes = [];
return view('teacher.presents', compact('students', 'classes'))->withErrors(['message' => 'Teacher not found.']);

Reply:
inizializzato le variabili per evitare di non averla quando viene fatta la return della view